File: apps/web/types/razorpay.ts
export interface RazorpayConnectPayload {
  key_id: string;
  key_secret: string;
}

export interface RazorpayVerifyResponse {
  valid: boolean;
  mode: "test" | "live";
}

export interface RazorpayStatusResponse {
  is_connected: boolean;
  mode: "test" | "live";
  key_id_masked: string | null;
  connected_at: string | null;
}

File: apps/backend/app/schemas/razorpay.py
from datetime import datetime
from pydantic import BaseModel, Field

class RazorpayConnectRequest(BaseModel):
    key_id: str = Field(min_length=8, max_length=100)
    key_secret: str = Field(min_length=8, max_length=100)

class RazorpayStatusResponse(BaseModel):
    is_connected: bool
    mode: str
    key_id_masked: str | None
    connected_at: datetime | None